datafusion/physical-plan/src/sorts/cursor.rs

```diff
@@ -40,6 +41,13 @@ pub trait CursorValues {
 
     /// Returns comparison of `l[l_idx]` and `r[r_idx]`
     fn compare(l: &Self, l_idx: usize, r: &Self, r_idx: usize) -> Ordering;
+
+    /// Slice at a given row index, returning a new Self
+    ///
+    /// # Panics
+    ///
+    /// Panics if the slice is out of bounds, or memory is insufficient
```

Comment:
I didn't see any code that would panic about memory being insufficient -- since this is a zero copy slice, I am not sure the memory check is needed

Reply:
Leftover from an earlier iteration. My error. TY for catching.